fe/fe-core/src/main/java/com/starrocks/analysis/ExportStmt.java

```diff
@@ -220,14 +222,45 @@ private void checkTable(Catalog catalog) throws AnalysisException {
                 case INLINE_VIEW:
                 case VIEW:
                 default:
-                    throw new AnalysisException("Table[" + tblName.getTbl() + "] is "
-                            + tblType.toString() + " type, do not support EXPORT.");
+                    throw new AnalysisException("Table[" + tblName.getTbl() + "] is " + tblType.toString() +
+                            " type, do not support EXPORT.");
+            }
+
+            if (partitions != null) {
+                for (String partitionName : partitions) {
+                    Partition partition = table.getPartition(partitionName);
+                    if (partition == null) {
+                        throw new AnalysisException("Partition [" + partitionName + "] does not exist.");
+                    }
+                }
             }
 
-            for (String partitionName : partitions) {
-                Partition partition = table.getPartition(partitionName);
-                if (partition == null) {
-                    throw new AnalysisException("Partition [" + partitionName + "] does not exist");
+            // check columns
+            if (columnNames != null) {
+                if (columnNames.isEmpty()) {
+                    throw new AnalysisException("Columns is empty.");
+                }
+
+                Map<String, Integer> tableColumnToIndex = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
+                int index = 0;
+                for (Column column : table.getBaseSchema()) {
+                    tableColumnToIndex.put(column.getName(), index++);
+                }
+                Set<String> uniqColumnNames = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+                int prevIndex = -1;
+                int currentIndex = -1;
+                for (String columnName : columnNames) {
+                    if (!uniqColumnNames.add(columnName)) {
+                        throw new AnalysisException("Duplicated column [" + columnName + "]");
+                    }
+                    if (!tableColumnToIndex.containsKey(columnName)) {
+                        throw new AnalysisException("Column [" + columnName + "] does not exist in table.");
+                    }
+                    currentIndex = tableColumnToIndex.get(columnName);
+                    if (prevIndex > currentIndex) {
+                        throw new AnalysisException("Columns order should be same with the schema.");
```

Comment:
Is this a necessary restriction?  this may confuse users.

Reply:
ok